app/controllers/etds_controller.rb

```diff
@@ -0,0 +1,163 @@
+# frozen_string_literal: true
+
+require 'resolv'
+
+# Controller for ETD endpoints used by Peoplesoft to transfer data to the ETD system
+class EtdsController < ApplicationController
+  skip_verify_authorized only: %i[index create]
+  before_action :authenticate, only: %i[index create], unless: proc { request_from_authorized_origin? }
+  before_action :set_submission, only: %i[create]
+
+  PHD_REGEX = /p\W*h\W*d/i
+  ENG_REGEX = /^eng$/i
+
+  attr_reader :invalid_xml_message, :message, :submission
+
+  # GET /etds
+  # only here for peoplesoft ping
+  def index
+    render html: 'OK'
+  end
+
+  # POST /etds
+  # receives xml data from peoplesoft
+  def create
+    logger.info("RAW XML: #{request.raw_post}")
+
+    return unless submission
+
+    @submission = RegisterService.register(submission:)
+    @submission.update!(submission_attributes)
+    peoplesoft_actions
+    # we regenerate signature pages in case new readers were added
+    # SignaturePageService.regenerate_signature_page(submission)
+    render_created
+  end
+
+  private
+
+  def dissertation_id
+    etd_params.expect(:dissertationid)
+  end
+
+  def etd_params
+    params.expect(DISSERTATION: [:dissertationid, :title, :type, :readerapproval, :readercomment, :regapproval,
+                                 :regcomment, :documentaccess, :schoolname, :degreeconfyr, :univid, :sunetid,
+                                 :readeractiondttm, :regactiondttm, :degree, :name, :vpname, :career, :program,
+                                 :plan,
+                                 { sub: [%i[deadline]] },
+                                 { subplan: [%i[code __content__]],
+                                   reader: [%i[sunetid name_prefix prefix name suffix type
+                                               univid readerrole finalreader]] }])
+  end
+
+  def title
+    etd_params[:title].gsub(/\s+/, ' ').strip
```

Comment:
Looks like it squishes multiple space characters into one? I wonder if we can use `String#squish` instead of the gsub.